chromadb/api/models/Collection.py

```diff
@@ -380,3 +381,26 @@ def delete(
             tenant=self.tenant,
             database=self.database,
         )
+
+
+
+class CollectionName(str):
+    """
+    A string wrapper to supply users with indicative message about list_collections only
+    returning collection names, in lieu of Collection object.
+    """
+
+    def __getattr__(self, name):
+        collection_attributes_and_methods = [
+            member for member, _ in inspect.getmembers(Collection)
+            if not member.startswith("_")
+        ]
+
+        if name in collection_attributes_and_methods:
+            raise NotImplementedError(
+                f"In Chroma v0.6.0, list_collections only returns collection names. "
+                f"Use get_collection to access Collection.{name}. "
+                f"See https://docs.trychroma.com/deployment/migration for more information."
+            )
+
+        raise AttributeError(f"'CollectionName' object has no attribute '{name}'")
```

Comment:
This implementation is a bit funky because `str` doesn't have a `__getattr__` but does have `__getattribute__` which is subtly different. I'm concerned that we would be shadowing `str` method calls accidentally;
- can we verify that this doesn't do anything unexpected with e.g. `.join`
- if it doesn't, can we comment on how this works at the level of method dispatch in each case?

Reply:
It is funky indeed. I can switch to explicit function definitions.

In any case this is a detailed description of the dispatch order:

When a user will try to access an attribute on a `CollectionName` string, the `__getattribute__` method of `str` is invoked first. If a valid str method or property is found, it will be used. Otherwise, the fallback `__getattr__` defined here is invoked next. It will error if the requested attribute is a `Collection` method or property.

For example:

```
collection_name = client.list_collections()[0] # collection_name = "test"
```

```
collection_name.startsWith("t") # Evaluates to True.
```

`__getattribute__`  is invoked first, selecting `startsWith` from `str`.

```
collection_name.add(ids=[...], documents=[...]) # Raises the error defined in this class.
```

`__getattribute__` is invoked first, not finding a match in `str`. `__getattr__` from this class is invoked and raises an error

Reply:
It's funky because if this was just `str`, `__getattr__` would never get called (and doesn't exist) - this is handled by `__getattribute__` checking for it. I think that's fine but we should make some sort of comment.